src/main/scala/com/sky/kafka/topicloader/config/config.scala

```diff
@@ -0,0 +1,56 @@
+package com.sky.kafka.topicloader
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+import cats.data.ValidatedNec
+import cats.implicits._
+import com.typesafe.config.{Config => TypesafeConfig, ConfigException}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.util.{Failure, Success, Try}
+
+package object config {
+  type ValidationResult[A] = ValidatedNec[Throwable, A]
+
+  implicit class TryOps[A](t: Try[A]) {
+    def validate(): ValidationResult[A] = t.toEither.toValidatedNec
+  }
+
+  final case class PosInt(private val _value: Int) {
+    val value: Int = _value
+  }
+
+  object PosInt {
+    def apply(value: Int): PosInt =
```

Comment:
can we return an either or a try instead of throwing in here? you end up wrapping it in try at the callsite anyway